content/cloud/contributing.md

```diff
@@ -60,24 +63,66 @@ Create a new remote for the upstream (a pointer to the original repository to wh
 $ git remote add upstream https://github.com/fermyon/developer
 ```
 
-## 5. Add Changes
+## 5. Code Blocks and Annotations
+
+It is highly recommended that you use either the `<!-- @selectiveCpy -->` or the `<!-- @nocpy -->` annotation before each of your code blocks, and that each code block defines the appropriate [syntax highlighting](https://rdmd.readme.io/docs/code-blocks#language-support). 
+
+The selective copy annotation (`<!-- @selectiveCpy -->`) is intended for use when communicating code and/or CLI commands for the reader to copy and paste. The selective copy annotation allows the reader to see the entire code block (both commands and results) but only copies the lines that start with `$` into the readers clipboard (minus the `$`) when the user clicks the copy button. For example, copying the following code block will only copy `echo "hello"` into your clipboard, for pasting.
+
+```bash
+$ echo "hello"
+hello
+```
+
+> Note: If the command, that starts with `$`, is deliberately spread over two lines (by escaping the newline character), then the copy mechanism will still copy the second line which is technically still part of that single command.
+
+The no copy annotation (`<!-- @nocpy -->`) preceeds a code block where no copy and pasting of code is intended. For example, use no copy when just displaying output or information to demonstrate a terminal's output.
+
+## 6. Check Content
+
+Once you are satisfied with your contribution, you can programmatically check your content.
+
+If you have not done so already, please go ahead and perform the `npm install` command; to enable Node dependencies such as `markdownlint-cli2`.
+
+<!-- @selectiveCpy -->
+
+```bash
+$ cd ~
+$ cd developer
+$ npm install
+```
+With all Node dependencies installed, you can now check for broken links and also lint your markdown files using the following `test` command:
+
+<!-- @selectiveCpy -->
 
-Once you are satisfied with your contribution go ahead and add your changes by moving to a top-level directory, under which your changes exist i.e. `cd ~/developer`.
+```bash
+$ cd ~
+$ cd developer
+$ npm test
+```
+
+## 7. Add Changes
+
+Once your changes have been checked, go ahead and add your changes by moving to a top-level directory, under which your changes exist i.e. `cd ~/developer`.
 
 Add your changes using the following command:
 
 <!-- @selectiveCpy -->
 
 ```bash
+$ cd ~
+$ cd developer
 $ git add
 ```
 
-## 6. Commit Changes
+## 7. Commit Changes
 
 Before committing, please ensure that your GitHub installation is configured sufficiently so that you can `--signoff` as part of the `git commit` command. For example, please ensure that the `user.name` and `user.email` are configured in your terminal. You can check if these are set by typing `git config --list`.
 
 If you need to set these values please use the following commands:
 
+<!-- @selectiveCpy -->
+
 ```bash
 $ git config user.name "yourusername"
 $ git config user.email "youremail@somemail.com"
```

Comment:
Might want to split this into 2 code blocks as the user needs to modify each of the statements.

Reply:
Done.